tests/integration_tests/ui_tests/ArelleGUITest/ArelleGUITest/Tests.cs

```diff
@@ -0,0 +1,362 @@
+                var pythonExe = Environment.GetEnvironmentVariable("ARELLE_PYTHON_EXE") ?? Path.Combine(arellePath!, "venv", "Scripts", "python.exe");
+                ExecutionType = new ExecutionTypeT.Source(
+                    ArellePath: arellePath,
+                    ArelleResourcesPath: arelleResourcesPathEnv ?? Path.GetFullPath(Path.Join(up, @"tests\integration_tests\ui_tests\resources")),
+                    UseVirtualEnv: inTree,
+                    PythonExe: pythonExe);
+            }
+            else
+            {
+                if (arelleResourcesPathEnv == null)
+                {
+                    throw new ArgumentNullException("ARELLE_RESOURCES_PATH", "must be specified for non-source runs");
+                }
+                ExecutionType = new ExecutionTypeT.Build(ArellePath: arellePathEnv, ArelleResourcesPath: arelleResourcesPathEnv);
+            }
+        }
+
+        protected override AutomationBase GetAutomation()
+        {
+            Logger.Default.SetLevel(LogLevel.Debug);
+            var automation = new UIA3Automation
+            {
+                ConnectionTimeout = TimeSpan.FromSeconds(10),
+                TransactionTimeout = TimeSpan.FromSeconds(10),
+            };
+            return automation;
+        }
+
+        protected override Application StartApplication()
+        {
+            Directory.CreateDirectory(TestsMediaPath);
+            Application app;
+            switch (ExecutionType)
+            {
+            case ExecutionTypeT.Source executionType:
+            {
+                var arelleGUIPyPath = Path.Combine(executionType.ArellePath!, "arelleGUI.pyw");
+                var parentProcess = Process.Start(
+                    "cmd", $"/c \"{executionType.PythonExe} {arelleGUIPyPath} > {TestsMediaPath}\\{TestContext.CurrentContext.Test.MethodName}-log.txt 2>&1\"");
+
+                Process? MaybeFindProcess()
+                {
+                    foreach (var p in Process.GetProcesses())
+                    {
+                        var isPython = p.ProcessName.EndsWith("python", StringComparison.OrdinalIgnoreCase);
+                        if (!isPython) continue;
+                        var ppid = GetParentProcessId(p);
+                        if (ppid == null) continue;
+                        if (executionType.UseVirtualEnv)
+                        {
+                            // virtualenv python creates a child python process, so look at the grandparent pid
+                            ppid = GetParentProcessId(Process.GetProcessById(ppid.Value));
+                        }
+                        if (ppid == parentProcess.Id)
+                        {
+                            return p;
+                        }
+                    }
+                    return null;
+                }
+                var process = Retry.WhileNull(MaybeFindProcess, timeout: TimeSpan.FromMinutes(2), interval: TimeSpan.FromMilliseconds(500)).Result;

```

Comment:
Suggest constants, and maybe even env vars for all time configuration

Reply:
> Creating two constants for every retry seems excessive and makes you have to navigate/hover to read the code.

Sure, but it also documents what the timeout is, and lets you mess with values if something starts failing randomly without having to read the code to understand.